### DISCUSSION TITLE
Having problem download Azurite Docker Container

### DISCUSSION THREAD

**Discussion by gkarwchan** [Azure/Azurite#2522] [Q&A]
I asked this questions on Docker forum, and on stackoverflow and didn't get help, and I am hoping I can get help here.
I am using Azurite as a docker container.
I am using Docker Desktop for windows 11 with WSL 2 integration, and running it as Linux containers.

I have 7 images/containers including Azurite (ProgressDB, ElasticSearch, Redis, RabbitMq,... and Azurite).
I have been running those containers for more than 3 years, and always keep updating Docker desktop for latest version, without any problem.

But on last time I upgraded Docker desktop from 4.36 to 4.37.1.
The upgrade ereased existing containers, so I rum my docker-compose to re-run my containers.
All of them run ok, except Azurite.
The image itself is not downloading.
when I do 
`docker pull mcr.microsoft.com/azure-storage/azurite
`
It returns this error message

`Error response from daemon: failed to resolve reference "mcr.microsoft.com/azure-storage/azurite:latest": failed to do request: Head "https://mcr.microsoft.com/v2/azure-storage/azurite/manifests/latest": EOF
`
Something on my machine is preventing docker from accessing that URL.
My machine is a personal machine, so there is no proxy setting, or VPN, or anything like that.

I tried to acess that URL in the error message: https://mcr.microsoft.com/v2/azure-storage/azurite/manifests/latest
And I can access it correctly from my browser or from "curl"

I ping mcr.microsoft.com, and it is ping correctly.
I downgrade Docker Desktop to its previous version, but the problem still the sam.

I tried to clear all my docker cache by doing system prune

`docker system prune --all
`

and re-pull and re-run all my images and containers, and all of them (6 others) running ok, except Azurite.

hopefully someone can help me

**Comment by blueww** [comment]
@gkarwchan 

I can't repro you issue. 
I am also on docker desktop 4.37.1, and also use Linux container. I also use Win 11.
I just tested by remove all Azurite docker image and run "docker pull mcr.microsoft.com/azure-storage/azurite", the latest azurite image can be downloaded successfully.

Not sure why this happen on you machine.

Would you like to install Azurite with npm and run it as a workaround?
If you have Visual studio installed on your machine, you might can also run Azurite in VS, or run Azurite exe in commandline without open VS. https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azurite?tabs=visual-studio%2Cblob-storage#running-azurite-from-the-command-line

![image](https://github.com/user-attachments/assets/7b608ed1-66ab-47f4-bb37-6d7e6bd5f394)

**Reply by gkarwchan** [reply]
Yes, it is not an issue with Azurite docker image itself. 
It is an issue on my machine and specifically Docker server communicating to that URL address.

and it is not only that image, but all images from microsoft (including .NET sdk, and others)

I asked this question here as a dart in the dark and see if anybody can guide me how to debug the issue.

**Reply by curtferris** [reply]
Hey @gkarwchan,

I am based in Australia and I have been having the same issues on my PC and have tried everything. Then I had a bit of a thought and used my VPN and set it to be San Francisco. Then it worked fine. I am guessing that there is a likely an issue on the CDN or something for docker hub. If you have a VPN I suggest you just set to the USA for the moment.

**Comment by leadsoftlucas** [comment]
From Brazil, I am facing the same issue.
